contracts/collect/StepwiseCollectModule.sol

```diff
@@ -0,0 +1,283 @@
+            });
+            return abi.encode(initData);
+        }
+    }
+
+    /**
+     * @dev Processes a collect by:
+     *  1. Ensuring the collector is a follower
+     *  2. Ensuring the current timestamp is less than or equal to the collect end timestamp
+     *  3. Ensuring the collect does not pass the collect limit
+     *  4. Charging a fee
+     */
+    function processCollect(
+        uint256 referrerProfileId,
+        address collector,
+        uint256 profileId,
+        uint256 pubId,
+        bytes calldata data
+    ) external override onlyHub {
+        if (_dataByPublicationByProfile[profileId][pubId].followerOnly)
+            _checkFollowValidity(profileId, collector);
+        uint256 endTimestamp = _dataByPublicationByProfile[profileId][pubId].endTimestamp;
+        if (endTimestamp != 0 && block.timestamp > endTimestamp) revert Errors.CollectExpired();
+
+        if (
+            _dataByPublicationByProfile[profileId][pubId].collectLimit != 0 &&
+            _dataByPublicationByProfile[profileId][pubId].currentCollects >=
+            _dataByPublicationByProfile[profileId][pubId].collectLimit
+        ) {
+            revert Errors.MintLimitExceeded();
+        } else {
+            ++_dataByPublicationByProfile[profileId][pubId].currentCollects;
+            if (referrerProfileId == profileId) {
+                _processCollect(collector, profileId, pubId, data);
+            } else {
+                _processCollectWithReferral(referrerProfileId, collector, profileId, pubId, data);
+            }
+        }
+    }
+
+    /**
+     * @notice Returns the publication data for a given publication, or an empty struct if that publication was not
+     * initialized with this module.
+     *
+     * @param profileId The token ID of the profile mapped to the publication to query.
+     * @param pubId The publication ID of the publication to query.
+     *
+     * @return ProfilepublicationData The ProfilePublicationData struct mapped to that publication.
+     */
+    function getPublicationData(uint256 profileId, uint256 pubId)
+        external
+        view
+        returns (ProfilePublicationData memory)
+    {
+        return _dataByPublicationByProfile[profileId][pubId];
+    }
+
+    // TODO: Decide if we want keep it "public" or downgrade to "internal"?
+    // Pro-public: Easier frontend calculation/verification of how much you need to pay
+    // Cons-public: extra function call - more gas
+    function calculateFee(ProfilePublicationData memory data) public pure returns (uint256) {
+        // Because we already incremented the current collects in storage - we need to adjust it here.
+        // This is done to allow the first collect price to be equal to c parameter (better UX)
+        uint256 collects = data.currentCollects - 1;
+        // TODO: Probably unnecessary optimization - verify if it's necessary:
+        if (data.a == 0 && data.b == 0) return data.c;
```

Comment:
This line might be unnecessary optimization, just because it would only apply if publishers use this module for a flat, constant fee on collects (independent of the number of collects so far). This almost seems like a misuse of the Lens system when there would be gas-cheaper and simpler FeeCollect modules that suit this use case.

```suggestion
```

Reply:
I've left the optimization cause it was still saving gas

Reply:
True. When this was designed we didn't have the universal FeeCollect module, so allowing constant price (when a & b = 0) looked like a good idea. Now probably rare (or never used) usecase - so considering to remove this entirely.